src/core/load-fragment.c

```diff
@@ -6489,3 +6491,58 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_log_filter_patterns(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        ExecContext *c = ASSERT_PTR(data);
+        _cleanup_(pattern_freep) pcre2_code *compiled_pattern = NULL;
+        const char *pattern = ASSERT_PTR(rvalue);
+        bool is_allowlist = true;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+
+        if (isempty(pattern)) {
+                /* Empty assignment resets the lists. */
+                c->log_filter_allowed_patterns = set_free(c->log_filter_allowed_patterns);
+                c->log_filter_denied_patterns = set_free(c->log_filter_denied_patterns);
+                return 0;
+        }
+
+        if (pattern[0] == '~') {
+                is_allowlist = false;
+                pattern++;
+                if (isempty(pattern))
+                        /* LogFilterPatterns=~ is not considered a valid pattern. */
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                          "Regex pattern invalid, ignoring: %s=%s", lvalue, rvalue);
+        }
+
+        r = pattern_compile_and_log(pattern, 0, &compiled_pattern);
+        if (r < 0) {
+                if (r == -ENOMEM)
+                        return log_oom();
+                return 0;
+        }
```

Comment:
`log_oom()` can be also dropped. I do not think we should check if the error code is ENOMEM or not:
```c
if (pattern_compile_and_log(...) < 0)
        return 0;
```
If you want to check ENOMEM here, then please also check ENOMEM by `set_put_strdup()` below.